src/qibolab/pulses/pulse.py

```diff
@@ -21,29 +19,29 @@ class PulseType(Enum):
     DRIVE = "qd"
     FLUX = "qf"
     COUPLERFLUX = "cf"
+    DELAY = "dl"
+    VIRTUALZ = "virtual_z"
 
 
 @dataclass
 class Pulse:
-    """A class to represent a pulse to be sent to the QPU."""
+    """A pulse to be sent to the QPU."""
 
-    start: int
-    """Start time of pulse in ns."""
     duration: int
     """Pulse duration in ns."""
     amplitude: float
     """Pulse digital amplitude (unitless).
 
     Pulse amplitudes are normalised between -1 and 1.
     """
-    frequency: int
+    frequency: int = 0
```

Comment:
Is it useful to have a default here?

Reply:
I think this was to simplify deserialization, because I am using https://github.com/qiboteam/qibolab/blob/bd4ab21a1571357f0cb778c506520660ce8ccac9/src/qibolab/serialize.py#L94
where `pulse_kwargs` is the `dict` loaded from the platform runcard json and for some pulses like flux, this does not specify frequency. Of course, other options are to correct for it in serialize.py, or force specifying frequency in all runcards (less convenient) but I found simpler just putting a default value. Let me know if you disagree.

Reply:
I see your point, but I'd rather suggest to have an alternative constructor for flux pulses, as we have:
https://github.com/qiboteam/qibolab/blob/bd4ab21a1571357f0cb778c506520660ce8ccac9/src/qibolab/pulses/pulse.py#L71-L73
and then serialize and deserialize them as normal pulses. At least there are fewer exceptional cases around, and if you want to omit arguments you have to say why (that's the sense of the alternative constructor).

Moreover, if flux pulses did not have a frequency, I would have put `Optional[int]` as type. But they have a frequency, since they are just continuous pulses. Thus `int` (or `float`) is truly correct, we need no other special value :)